05-pagination-querying.md

```diff
@@ -0,0 +1,366 @@
+# Pagination, Querying and Sorting
+
+## What is pagination?
+
+Pagination of data is a way to limit the output of a list of resources to a well-defined (maximum) size.
+This paginated list is provided to the consumer of the REST api as a result of a query on a certain resource.
+
+Instead of returning a complete list of records, only a fraction of the data is returned with information necessary to retrieve the rest of the data.
+The result is often wrapped in an envelope that contains the following data:
+* The number of the current page
+* The total number of pages
+* The content of this page, which is a list of elements.
+
+## Why paginate data?
+Often the user of a web application is not interested in hundreds of resources.
+During a search operation for example they typically specify some filters to the best of their knowledge so
+the best results end up on top of the page.
+
+Because the user is typically not interested in most of the elements present on the server, pagination (in combination with a good search function)
+can provide them the data they need without sending data over the line.
```

Comment:
Getting data without sending data over the line? Sounds like a very nice compression algorithm that could make us very richt haha :P